### PR TITLE
[MER-2595] Enable discussion activity for all pages

### PR DESCRIPTION
**Pull request by nicocirio**
Link to the ticket

### Enabling Student Course Portal Collaborative Space

https://github.com/Simon-Initiative/oli-torus/assets/74839302/b23600a1-1630-4f7b-af9f-ce0664482550

### Manually enabling one page collab space (to check the collab space count)

https://github.com/Simon-Initiative/oli-torus/assets/74839302/d596c1a2-53d7-438e-a410-c9bfd8287059

### Enabling Collab Spaces for all pages in the course
Note that the confirmation modal renders a form to provide the collab space configuration that will be bulk-applied to all pages.

https://github.com/Simon-Initiative/oli-torus/assets/74839302/616b14eb-6b10-4ada-9273-339e8d332a71

### Disabling Collab Spaces for all pages in the course

https://github.com/Simon-Initiative/oli-torus/assets/74839302/7bc6209b-fe67-4518-8143-583e2f81f2fc

### CODE REVIEW

File: lib/oli/resources/collaboration.ex [hunk 1]
```diff
@@ -254,6 +254,160 @@ defmodule Oli.Resources.Collaboration do
...
+          )
+        ),
+        count(rev)
+      }
+    )
+    |> Repo.one()
+  end
+
+  @doc """
+  Counts the number of pages with collab space enabled and the number of total pages for a given project,
```
Comment (gastonabella) [inline lib/oli/resources/collaboration.ex:300]: Nit: _"...for a given **section**."_

File: lib/oli/resources/collaboration.ex [hunk 2]
```diff
@@ -254,6 +254,160 @@ defmodule Oli.Resources.Collaboration do
...
+  def count_collab_spaces_enabled_in_pages_for_project(project_slug) do
+    page_id = Oli.Resources.ResourceType.get_id_by_type("page")
+
+    project_working_publication =
+      from(p in Publication,
+        join: c in Project,
+        on: p.project_id == c.id,
+        where: is_nil(p.published) and c.slug == ^project_slug,
+        select: p.id
+      )
```
[the remark this answers is not in the document]
Comment (nicocirio) [reply lib/oli/resources/collaboration.ex]: makes sense!

**Review by gastonabella — APPROVED** [review]
Looks great!

File: lib/oli_web/live/collaboration_live/collab_space_config_view.ex [hunk 3]
```diff
@@ -289,6 +437,21 @@ defmodule OliWeb.CollaborationLive.CollabSpaceConfigView do
...
+
+  defp get_collab_space_pages_count(false, project_slug) do
+    Collaboration.count_collab_spaces_enabled_in_pages_for_project(project_slug)
+  end
+
+  defp get_collab_space_pages_count(true, section_slug) do
+    Collaboration.count_collab_spaces_enabled_in_pages_for_section(section_slug)
+  end
+
+  defp get_collab_space_pages_count(nil, _), do: %{with_collab_spaces_enabled: 0, total: 0}
```
[the remark this answers is not in the document]
Comment (nicocirio) [reply lib/oli_web/live/collaboration_live/collab_space_config_view.ex]: You are right, I removed it!

**Review by darrensiegel — APPROVED** [review]

**Pull request merged**